### ISSUE TITLE
conll_learn.lua:60: attempt to concatenate field 'Rec' (a nil value)

### ISSUE THREAD

**Issue by SeekPoint** [sld/torch-conv-ner#2]
}
union set size is 	0	6199	6199	
version	1-convolution-sentence-net-fixed	21	
model_name	convolution-full		
cuda	t	false	
1.000 percent complete; 0.019 sec per batch; 0.196 batch err;
Epoch: 1	TrainLoss: 0.54399251100247	EpochTime: 148.87882399559
	
Eval set loss: nan	
Time: 1.0967254638672e-05	
processed 0 tokens with 0 phrases; found: 0 phrases; correct: 0.	
/home/rzai/torch/install/bin/luajit: conll_learn.lua:60: attempt to concatenate field 'Rec' (a nil value)
stack traceback:
	conll_learn.lua:60: in function 'on_train'
	conll_learn.lua:188: in function 'train_model'
	conll_learn.lua:293: in function 'run'
	conll_learn.lua:296: in main chunk
	[C]: in function 'dofile'
	...rzai/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
	[C]: at 0x00406670
rzai@rzai00:~/prj/torch-conv-ner$

**Comment by SeekPoint** [comment]
rzai@rzai00:~/prj/torch-conv-ner$ bash experiments/convolution-net.sh
cat: data/conll2003/eng.testa.dev: No such file or directory
cat: data/conll2003/eng.testb.test: No such file or directory
('LOC', ' max ngram ', 'data/gazetteers/ner.loc.lst', 9)
('MISC', ' max ngram ', 'data/gazetteers/ner.misc.lst', 13)
('PER', ' max ngram ', 'data/gazetteers/ner.per.lst', 11)
('ORG', ' max ngram ', 'data/gazetteers/ner.org.lst', 8)
Processing eng.testa.dev.iobes with sentence-convolution mode
(0, 0)
Processing eng.train.iobes with sentence-convolution mode
(946, 203621)
Processing eng.testb.test.iobes with sentence-convolution mode
(0, 0)
{
  isCapDataset : false
  mode : "win"
  comprenoInFile : ""
  batchedMode : false
  len : "50"
  outFile : "data/embeddings/senna.torch"
  comprenoOutFile : ""
  inFile : "data/embeddings/senna.index"
}
num input lines = 15704	
{
  isCapDataset : false
  mode : "sentence-convolution"
  comprenoInFile : ""
  batchedMode : false
  len : ""
  outFile : "data/conll2003/eng.testb.test.iobes.torch"
  comprenoOutFile : ""
  inFile : "data/conll2003/eng.testb.test.iobes.index"
}
num input lines = 0	
{
  isCapDataset : false
  mode : "sentence-convolution"
  comprenoInFile : ""
  batchedMode : false
  len : ""
  outFile : "data/conll2003/eng.testa.dev.iobes.torch"
  comprenoOutFile : ""
  inFile : "data/conll2003/eng.testa.dev.iobes.index"
}
num input lines = 0	
{
  isCapDataset : false
  mode : "sentence-convolution"
  comprenoInFile : ""
  batchedMode : true
  len : ""
  outFile : "data/conll2003/eng.train.iobes.torch"
  comprenoOutFile : ""
  inFile : "data/conll2003/eng.train.iobes.index"
}
num input lines = 203621	
203621	203621	6199	
{
  with_separate_lr : false
  mode : "sent"
  version : "1-convolution-sentence-net-fixed"
  batch_size : 1
  model_name : "convolution-full"
  useOldNetParams : false
  cuda : true
  dump : ""
}
using GPU	
nn.Sequential {
  [input -> (1) -> (2) -> (3) -> (4) -> (5) -> (6) -> (7) -> (8) -> (9) -> (10) -> output]
  (1): nn.SplitTable
  (2): nn.ParallelTable {
    input
      |`-> (1): nn.LookupTable
      |`-> (2): nn.LookupTable
      |`-> (3): nn.LookupTable
      |`-> (4): nn.LookupTable
      |`-> (5): nn.LookupTable
      |`-> (6): nn.LookupTable
       `-> (7): nn.LookupTable
       ... -> output
  }
  (3): nn.JoinTable
  (4): nn.TemporalConvolution
  (5): nn.Max
  (6): nn.Dropout(0.500000)
  (7): nn.Linear(300 -> 300)
  (8): nn.Dropout(0.500000)
  (9): nn.HardTanh
  (10): nn.Linear(300 -> 17)
}
union set size is 	0	6199	6199	
version	1-convolution-sentence-net-fixed	21	
model_name	convolution-full		
cuda	t	false	
1.000 percent complete; 0.019 sec per batch; 0.196 batch err;
Epoch: 1	TrainLoss: 0.54399251100247	EpochTime: 148.87882399559
	
Eval set loss: nan	
Time: 1.0967254638672e-05	
processed 0 tokens with 0 phrases; found: 0 phrases; correct: 0.	
/home/rzai/torch/install/bin/luajit: conll_learn.lua:60: attempt to concatenate field 'Rec' (a nil value)
stack traceback:
	conll_learn.lua:60: in function 'on_train'
	conll_learn.lua:188: in function 'train_model'
	conll_learn.lua:293: in function 'run'
	conll_learn.lua:296: in main chunk
	[C]: in function 'dofile'
	...rzai/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
	[C]: at 0x00406670
rzai@rzai00:~/prj/torch-conv-ner$ 
rzai@rzai00:~/prj/torch-conv-ner$ git remote -v
origin	https://github.com/sld/torch-conv-ner.git (fetch)
origin	https://github.com/sld/torch-conv-ner.git (push)
rzai@rzai00:~/prj/torch-conv-ner$

**Comment by sld** [comment]
Seems that you did not put CoNLL files properly.
They should placed like described in README https://github.com/sld/torch-conv-ner#preparation:
- data/conll2003/eng.testa.dev;
- data/conll2003/eng.testb.test;
- data/conll2003/eng.train

**Comment by SeekPoint** [comment]
how to generate

data/conll2003/eng.testa.dev;
data/conll2003/eng.testb.test;
data/conll2003/eng.train
from  https://github.com/patverga/torch-ner-nlp-from-scratch/tree/master/data/conll2003, I see:
should I rename some file name?

**Comment by SeekPoint** [comment]
I think it can fix by:

rzai@rzai00:~/prj/torch-conv-ner/data/conll2003$ cp ../../../torch-ner-nlp-from-scratch/data/conll2003/eng.testa eng.testa.dev
rzai@rzai00:~/prj/torch-conv-ner/data/conll2003$ cp ../../../torch-ner-nlp-from-scratch/data/conll2003/eng.testb eng.testb.test
rzai@rzai00:~/prj/torch-conv-ner/data/conll2003$ cp ../../../torch-ner-nlp-from-scratch/data/conll2003/eng.train eng.train
rzai@rzai00:~/prj/torch-conv-ner/data/conll2003$
